### ISSUE TITLE
[Feature] use io_ring for file io?

### ISSUE THREAD

**Issue by power1628** [apache/incubator-doris#7001]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, Doris is using a buffered io interface to read/write files.
Any plan try to leverage Linux io_ring feature to optimize file io?

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by morningman** [comment]
You mean io_uring? Looks like it needs a higher linux kernal version(5.0+)?

**Comment by power1628** [comment]
@morningman  yes, io_uring(sorry for typo) and higher linux kernel version. There's one PR from rocksdb community, using io_uring to improve performance. https://github.com/facebook/rocksdb/pull/5881

**Comment by morningman** [comment]
> @morningman yes, io_uring(sorry for typo) and higher linux kernel version. There's one PR from rocksdb community, using io_uring to improve performance. [facebook/rocksdb#5881](https://github.com/facebook/rocksdb/pull/5881)

Thank, but at present, the kernel version in most of our environments has not reached 5.0, so this work may need to be completed later.
